lib/pages/ui/nps_rating_bar.dart

```diff
@@ -2,28 +2,42 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
 const String RATING_TYPE_NPS = "nps";
+const int DEFAULT_SELECTED_RATE = 4;
 
 class NpsRatingBar extends StatefulWidget {
-  final int counter;
+  final List<String> ids;
   final String minTitle;
   final String maxTitle;
+  final Function(Map<String, String>) onRatingListener;
 
-  NpsRatingBar({Key key, this.counter, this.minTitle, this.maxTitle})
+  NpsRatingBar(
+      {Key key,
+      this.ids,
+      this.minTitle,
+      this.maxTitle,
+      @required this.onRatingListener})
       : super(key: key);
 
   @override
   _NpsRatingBarState createState() => _NpsRatingBarState();
 }
 
 class _NpsRatingBarState extends State<NpsRatingBar> {
-  int _selectedRate = 4;
+  int _selectedRate = DEFAULT_SELECTED_RATE;
 
   void _onRateSelected(int rate) {
     setState(() {
       _selectedRate = rate;
+      widget.onRatingListener.call({widget.ids[rate]: ""});
     });
   }
 
+  @override
+  void didChangeDependencies() {
+    widget.onRatingListener.call({widget.ids[DEFAULT_SELECTED_RATE]: ""});
```

Comment:
I think you are going to set the default answer when the widget is built for the first time. I think `initState` would be the correct place since `didChangeDependencies` is called each time the widget is rebuilt. Please correct me if I'm wrong 🙇

Reply:
actually `didChangeDependencies` will be called right after `initState` it seems, and it's guaranteed to be called once a State is initialized as well, they are just slightly different on the lifecycle order. But `initState` does make more sense, so I will use it here.
![image](https://user-images.githubusercontent.com/13435717/126286583-d6c7891e-b3df-49f4-9ff5-6fff399ff7ec.png)